test/e2e/apps/cronjob.go

```diff
@@ -456,7 +466,7 @@ var _ = SIGDescribe("CronJob", func() {
 		cjs, err = cjClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
 		// Should have <= 2 items since some cronjobs might not have been deleted yet due to finalizers
-		framework.ExpectEqual(len(cjs.Items) <= 2, true, "filtered list should be <= 2")
+		gomega.Expect(len(cjs.Items)).To(gomega.BeNumerically("<=", 2), fmt.Sprintf("list of CronJob should should have <= 2 items, but got %d", len(cjs.Items)))
```

Comment:
Here the numeric comparison makes sense. Because Gomega already prints the actual length, IMHO it makes more sense to print the items in the custom message:

```suggestion
		gomega.Expect(len(cjs.Items)).To(gomega.BeNumerically("<=", 2), fmt.Sprintf("list of CronJob should should have <= 2 items, but got %+v", cjs.Items))
```

Reply:
Or, for the sake of consistency, always use `if len(...)` and a custom `framework.Failf`.

Reply:
I choose this way: `if len(...)` and a custom `framework.Failf`